cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestination.java

```diff
@@ -949,9 +949,16 @@ public DefaultHttpDestination build()
 
             // handle proxy type == OnPremise
             if( builder.get(DestinationProperty.PROXY_TYPE).contains(ProxyType.ON_PREMISE) ) {
-                final DefaultHttpDestination proxyDestination = proxyHandler.handle(this);
-                if( proxyDestination != null ) {
-                    return proxyDestination;
+                // Any change here must be copied to the HttpClientWrapper constructor
+                final Try<DefaultHttpDestination> proxyDestination = Try.of(() -> proxyHandler.handle(this));
+                if( proxyDestination.isSuccess() ) {
+                    return proxyDestination.get();
+                }
+                if( proxyDestination.isFailure() ) {
+                    log
+                        .error(
+                            "Unable to resolve proxy configuration for destination. This destination cannot be used for anything other than reading its properties.",
+                            proxyDestination.getCause());
```

Comment:
Unnecessary if condition:

```suggestion
                final String msg =
                    "Unable to resolve proxy configuration for destination. This destination cannot be used for anything other than reading its properties.";
                log.error(msg, proxyDestination.getCause());
```